### DISCUSSION TITLE
Is there a way to use the authenticated user in a GraphQL query or mutation?

### DISCUSSION THREAD

**Discussion by christopherliu** [keystonejs/keystone#7514] [Questions] [answered]
I have a frontend app where I'd like to log user actions in a list that includes the id of the current user, like this:

```graphql
        createUserAction(
            data: {
                user: {
                    connect: {
                        id: $user_id
                    }
                }
                path: $path
                action_name: $action_name
                parameters: $parameters
            }
        ) {
            id
        }
```

The problem is that the app has to make two round trips -- first using `authenticatedItem `to get the user's id, and then again to log the user action. This is troublesome in particular because not only is it a headache to ensure the user id is already retrieved in my frontend Next.JS app, but also it's a security hole if I can pass any user id to this query, so I'd prefer that the query defaults to the currently logged in user.

Is there a way to get the current user ID in a GraphQL query? Thanks!

**Comment by dcousens** [comment, marked as answer]
Hi @christopherliu,

This is not possible at this time.

We do have a feature request for this feature in https://github.com/keystonejs/keystone/discussions/7248.
Please follow up with any questions/suggestions in that discussion.
